src/shortestpaths/yen.jl

```diff
@@ -0,0 +1,98 @@
+"""
+    struct YenState{T, U}
+Designed for yen k-shortest-paths calculations.
+"""
+struct YenState{T, U<:Integer}<: AbstractPathState
+    dists::Vector{T}
+    paths::Vector{Vector{U}}
+end
+
+
+"""
+    yen_k_shortest_paths(g, source, target, distmx=DefaultDistance(), K=1);
+
+    Perform [Yen's algorithm](http://en.wikipedia.org/wiki/Yen%27s_algorithm)
+    on a graph, computing k-shortest distances between `source` and `target` other vertices.
+    Return a [`YenState`](@ref) that contains distances and paths.
+"""
+function yen_k_shortest_paths(
+    g::AbstractGraph,
+    source::U,
+    target::U,
+    distmx::AbstractMatrix{T}=DefaultDistance(),
+    K::Int=1) where T <: Real where U<:Integer
+
+    if source == target
+        return YenState{T, U}([U(0)], [[source]])
+    end
+
+    dj = dijkstra_shortest_paths(g, source, distmx)
+    path = enumerate_paths(dj)[target]
+    if isempty(path)
+        return YenState{T, U}(Vector{T}(), Vector{Vector{U}}())
+    end
+
+    dists = Array{T,1}()
+    push!(dists,dj.dists[target])
+    A = [path]
+    B = PriorityQueue()
+    gcopy = deepcopy(g)
+
+    for k = 1:(K-1)
+        for j = 1:length(A[k])
+            # Spur node is retrieved from the previous k-shortest path, k − 1
+            spurnode = A[k][j]
+            #  The sequence of nodes from the source to the spur node of the previous k-shortest path
+            rootpath = A[k][1:j]
+
+            # Store the removed edges
+            edgesremoved = Array{Tuple{Int,Int},1}()
+            # Remove the links of the previous shortest paths which share the same root path
+            for ppath in A
+                if length(ppath) > j && rootpath == ppath[1:j]
+                    u = ppath[j]
+                    v = ppath[j + 1]
+                    if has_edge(gcopy, u, v)
+                        rem_edge!(gcopy, u, v)
+                        push!(edgesremoved,(u, v))
+                    end
+                end
+            end
+
+            # Remove node of root path
+            for n = 1:(length(rootpath)-1)
```

Comment:
spacing here, please:
```
for n = 1:(length(rootpath) - 1)
```
(there are a few places where spacing will improve readability. Basically, it's a good idea to follow every comma with a space, and surround operations with spaces (except for `=` in function signatures). All code about this point looks good (except for L57), but please check the rest of it.